File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/infrastructure/persistence/entities/ApiKeyEntity.java
package br.com.deltaglobalbank.identity.infrastructure.persistence.entities;

import java.time.Instant;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "api_keys", schema = "identity")
@SQLDelete(sql = "UPDATE identity.api_keys SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
public class ApiKeyEntity {

    @Id
    @Column(name = "id")
    private UUID id;

    @Column(name = "api_client_id", nullable = false)
    private UUID apiClientId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "key_hash", nullable = false)
    private String keyHash;

    @Column(name = "key_prefix", nullable = false)
    private String keyPrefix;

    @Column(name = "fingerprint")
    private String fingerprint;

    @Column(name = "expires_at")
    private Instant expiresAt;

    @Column(name = "last_used_at")
    private Instant lastUsedAt;

    @Column(name = "revoked_at")
    private Instant revokedAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "deleted_at")
    private Instant deletedAt;

    protected ApiKeyEntity() {
    }

    public ApiKeyEntity(
        UUID id,
        UUID apiClientId,
        String name,
        String keyHash,
        String keyPrefix,
        String fingerprint,
        Instant expiresAt,
        Instant lastUsedAt,
        Instant revokedAt,
        Instant createdAt,
        Instant updatedAt,
        Instant deletedAt
    ) {
        this.id = id;
        this.apiClientId = apiClientId;
        this.name = name;
        this.keyHash = keyHash;
        this.keyPrefix = keyPrefix;
        this.fingerprint = fingerprint;
        this.expiresAt = expiresAt;
        this.lastUsedAt = lastUsedAt;
        this.revokedAt = revokedAt;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.deletedAt = deletedAt;
    }

    public UUID getId() {
        return id;
    }

    public UUID getApiClientId() {
        return apiClientId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getKeyHash() {
        return keyHash;
    }

    public String getKeyPrefix() {
        return keyPrefix;
    }

    public String getFingerprint() {
        return fingerprint;
    }

    public Instant getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(Instant expiresAt) {
        this.expiresAt = expiresAt;
    }

    public Instant getLastUsedAt() {
        return lastUsedAt;
    }

    public void setLastUsedAt(Instant lastUsedAt) {
        this.lastUsedAt = lastUsedAt;
    }

    public Instant getRevokedAt() {
        return revokedAt;
    }

    public void setRevokedAt(Instant revokedAt) {
        this.revokedAt = revokedAt;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Instant updatedAt) {
        this.updatedAt = updatedAt;
    }

    public Instant getDeletedAt() {
        return deletedAt;
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/infrastructure/persistence/entities/ApiKeyEntity.kt
package br.com.deltaglobalbank.identity.infrastructure.persistence.entities

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.SQLDelete
import org.hibernate.annotations.SQLRestriction
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "api_keys", schema = "identity")
@SQLDelete(sql = "UPDATE identity.api_keys SET deleted_at = NOW() WHERE id = ?")
@SQLRestriction("deleted_at IS NULL")
class ApiKeyEntity(
    @Id
    @Column(name = "id")
    var id: UUID,

    @Column(name = "api_client_id", nullable = false)
    var apiClientId: UUID,

    @Column(name = "name", nullable = false)
    var name: String,

    @Column(name = "key_hash", nullable = false)
    var keyHash: String,

    @Column(name = "key_prefix", nullable = false)
    var keyPrefix: String,

    @Column(name = "fingerprint")
    var fingerprint: String,

    @Column(name = "expires_at")
    var expiresAt: Instant?,

    @Column(name = "last_used_at")
    var lastUsedAt: Instant?,

    @Column(name = "revoked_at")
    var revokedAt: Instant?,

    @Column(name = "created_at", nullable = false, updatable = false)
    var createdAt: Instant,

    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant,

    @Column(name = "deleted_at")
    var deletedAt: Instant? = null
)